queries.sql

```diff
@@ -105,4 +105,80 @@ HAVING COUNT(animals.id) = (
   SELECT MAX(animal_count) FROM (
     SELECT COUNT(*) AS animal_count FROM animals GROUP BY owners_id
   ) AS owner_animal_counts
-);
\ No newline at end of file
+);
+
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.name, visits.visit_date
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.visit_date DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT visits.animal_id)
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez';
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.name, species.name AS specialty
+FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id
+ORDER BY vets.name;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name, visits.visit_date
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+WHERE vets.name = 'Stephanie Mendez';
```

Comment:
- [ ] Kindly edit this query and make it only show the animal visited `between 1 April 2020 and 30 August 2020`.
[hint] It will be like that:
```sql
  SELECT animals.name, vistis.visit_date
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vets.name = 'Stephanie Mendez'
  AND visit_date > '2020-04-01' AND visit_date < '2020-08-30'
```